README.md

```diff
@@ -35,138 +56,74 @@ fatal: can't squash-merge: 'plugins/aem-assets-plugin' was never added
+      Medium: { minWidth: 768, maxWidth: 1023 },
+      Large: { minWidth: 1024, maxWidth: 9999 },
+    },
+    externalImageUrlPrefixes: [
+      ['https://delivery-p66302-e574366.adobeaemcloud.com/', createOptimizedPictureForDMOpenAPI],
+      ['https://s7ap1.scene7.com/is/image/varuncloudready/', createOptimizedPictureForDM],
+    ],
   };
 }
 ```
 
-You'd need to add the following code in `createOptimizedPicture` in `aem.js` to call the overidden version of this function
-```
-  if (window.hlx?.aemassets?.createOptimizedPicture) {
-    return window.hlx.aemassets.createOptimizedPicture(src, alt, eager, breakpoints);
-  }
-```
+The externalImageUrlPrefixes is a list of URL prefix-handler tuple pairs. For any image URL that starts with a matching prefix, the corresponding handler function (e.g., createOptimizedPictureForDMOpenAPI or createOptimizedPictureForDM) will be used to process that image
+
+You'd need to add the following code for handling external images inside decorateMain Function in `scripts.js` 
 
-Here's the complete code for `createOptimizedPicture` in `aem.js` with the above lines of code added
 ```
-function createOptimizedPicture(
-  src,
-  alt = '',
-  eager = false,
-  breakpoints = [{ media: '(min-width: 600px)', width: '2000' }, { width: '750' }],
-) {
-
-  // Add below lines of code //
-  if (window.hlx?.aemassets?.createOptimizedPicture) {
-    return window.hlx.aemassets.createOptimizedPicture(src, alt, eager, breakpoints);
-  }
-  // Add above lines of code //
-
-  const url = new URL(src, window.location.href);
-  const picture = document.createElement('picture');
-  const { pathname } = url;
-  const ext = pathname.substring(pathname.lastIndexOf('.') + 1);
-
-  // webp
-  breakpoints.forEach((br) => {
-    const source = document.createElement('source');
-    if (br.media) source.setAttribute('media', br.media);
-    source.setAttribute('type', 'image/webp');
-    source.setAttribute('srcset', `${pathname}?width=${br.width}&format=webply&optimize=medium`);
-    picture.appendChild(source);
-  });
-
-  // fallback
-  breakpoints.forEach((br, i) => {
-    if (i < breakpoints.length - 1) {
-      const source = document.createElement('source');
-      if (br.media) source.setAttribute('media', br.media);
-      source.setAttribute('srcset', `${pathname}?width=${br.width}&format=${ext}&optimize=medium`);
-      picture.appendChild(source);
-    } else {
-      const img = document.createElement('img');
-      img.setAttribute('loading', eager ? 'eager' : 'lazy');
-      img.setAttribute('alt', alt);
-      picture.appendChild(img);
-      img.setAttribute('src', `${pathname}?width=${br.width}&format=${ext}&optimize=medium`);
-    }
-  });
-
-  return picture;
+if (window.hlx.aemassets?.decorateExternalImages) {
+    window.hlx.aemassets.decorateExternalImages(main);
 }
 ```
 
-You'd need to add the following code in `loadBlock` in `aem.js` to call the overidden version of this function
```

Comment:
I think we do need to retain this for cases where the block is loaded later (not part of initial markup) and has to be decorated for external images.